x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -314,7 +314,7 @@ export const createLifecycleExecutor =
           [ALERT_WORKFLOW_STATUS]: alertData?.fields[ALERT_WORKFLOW_STATUS] ?? 'open',
           [EVENT_KIND]: 'signal',
           [EVENT_ACTION]: isNew ? 'open' : isActive ? 'active' : 'close',
-          [TAGS]: options.rule.tags,
+          [TAGS]: currentAlertData?.tags ?? options.rule.tags,
```

Comment:
I'm a little worried here about breaking existing behavior.  The way this is written, if there are ANY tags in the alert, they override what used to be written (`options.rule.tags`).  Don't we want to union both of these here?  (with a Set, to avoid duplicates).

Reply:
I was adding the rule tags to the alert in the executor, so here I used tags from alert first otherwise rule tags (because rule tags are already added in alert tags). As far as I have seen, `tags` list in alert was always empty, but of course if we are adding tags in alert somewhere else, this wouldn't work. I have modified this to use tags from both places with your suggestion above about using `Set`.